superset/charts/api.py

```diff
@@ -469,20 +471,21 @@ def data(self) -> Response:
             return self.response_400(
                 message=_("Request is incorrect: %(error)s", error=error.messages)
             )
+
+        # ACL control
         try:
             query_context.raise_for_access()
         except SupersetSecurityException:
             return self.response_401()
+
+        # get all query payload
         payload = query_context.get_payload()
         for query in payload:
             if query.get("error"):
                 return self.response_400(message=f"Error: {query['error']}")
-        result_format = query_context.result_format
-
-        response = self.response_400(
-            message=f"Unsupported result_format: {result_format}"
-        )
 
+        result_format = query_context.result_format
+        response = None
```

Comment:
Not sure if all errors are caught by marshmallow on this. Also setting `response = None` is dangerous has it may end up has a Flask response

@villebro any thought on this?

Reply:
this expression is unreachable `response = self.response_400(message=f"Unsupported result_format: {result_format}")`  , because marshmallow validate method can raise exception when request is unexpect.

can we move all try...catch express to marshmallow logic?

Reply:
@zhaoyongjie I've placed that backstop there to ensure so that if someone adds a new result type to the marshmallow schema but forgets to add it to the endpoint, they don't get a `None` result. Having said that, you're absolutely right that this method is extremely loaded and should not be this complex, with this type of logic preferably elsewhere (marshmallow schema, `QueryContext`/`QueryObject`). If we can make sure that there is no way of ending up at the end of this method with a `result_type` that is unsupported I'm ok with removing it. However, that probably requires moving said logic to `QueryContext`.